examples/hpc/dla-driver/src/lib.rs

```diff
@@ -1,11 +1,99 @@
 #![no_std]
 
-mod mmap;
+extern crate alloc;
 
+mod mmap;
+pub use mmap::{
+    DLA0_ADDR, MEMORY_BANK_0_OFFSET, MEMORY_BANK_10_OFFSET, MEMORY_BANK_11_OFFSET,
+    MEMORY_BANK_12_OFFSET, MEMORY_BANK_13_OFFSET, MEMORY_BANK_14_OFFSET, MEMORY_BANK_15_OFFSET,
+    MEMORY_BANK_1_OFFSET, MEMORY_BANK_2_OFFSET, MEMORY_BANK_3_OFFSET, MEMORY_BANK_4_OFFSET,
+    MEMORY_BANK_5_OFFSET, MEMORY_BANK_6_OFFSET, MEMORY_BANK_7_OFFSET, MEMORY_BANK_8_OFFSET,
+    MEMORY_BANK_9_OFFSET, MEMORY_BANK_BASE_ADDR,
+};
+
+use alloc::vec::Vec;
 use core::ptr;
-use headsail_bsp::sprint;
+use headsail_bsp::{sprint, sprintln};
 use mmap::*;
 
+pub struct Dla {
+}
+
+#[rustfmt::skip]
+pub enum MemoryBank {
```

Comment:
Consider making this Clone & Copy: `#[derive(Clone, Copy)]`. Integer sized types are cheap to copy and you won't have to bother with the borrow checker (i.e., `.clone` and ownership issues) in the user code.

Reply:
Added Clone & Copy, but not yet used anywhere.

Reply:
You'll notice it when you do something like this:

```rust
let dla = Dla::new();

let sel_bank = MemoryBank::Bank4;
dla.set_bank_attribute_1(sel_bank, attr);
dla.set_bank_attribute_2(sel_bank, attr2);
dla.set_bank_attribute_3(sel_bank, attr3);
```

Without `derive(Copy)` you'd need to `.clone()` on every `sel_bank` above.